### PR TITLE
 #127 Provide instructions for verifying release

### PR DESCRIPTION
**Pull request by mikewalch**
* Adding instructions for running integrations
* Will add more instructions for running example applications

### CODE REVIEW

File: pages/release-process.md [hunk 1]
```diff
@@ -144,6 +144,72 @@ When the vote passes on a release candidate, follow the steps below to complete
 
  8.  Send an email to `dev@fluo.apache.org` announcing new release.
 
+### Test a Fluo release
+
+Below are two common testing strategies to verify a Fluo release candidate:
```
Comment (ctubbsii) [inline pages/release-process.md]: Replace with less wordy, and more personal: "You can test a release by:"

File: pages/release-process.md [hunk 2]
```diff
@@ -144,6 +144,72 @@ When the vote passes on a release candidate, follow the steps below to complete
...
+
+1. Run the integration tests of projects that depend on Fluo (using the release candidate jars)
+
+1. Run example Fluo applications (using the release candidate tarball)
```
Comment (ctubbsii) [inline pages/release-process.md]: No need for both a bulleted list here, and then the sections below. It is redundant, and wordy. This bulleted list can go away completely.

File: pages/release-process.md [hunk 3]
```diff
@@ -144,6 +144,72 @@ When the vote passes on a release candidate, follow the steps below to complete
...
+
+These strategies are described below with step-by-step instructions.
```
Comment (ctubbsii) [inline pages/release-process.md]: The user can see that. There's no reason for this sentence.

File: pages/release-process.md [hunk 4]
```diff
@@ -144,6 +144,72 @@ When the vote passes on a release candidate, follow the steps below to complete
...
+
+#### Run integration tests
+
+A Fluo release can be tested by running the integration tests of projects that use Fluo.
```
Comment (ctubbsii) [inline pages/release-process.md]: This sentence isn't necessary.

File: pages/release-process.md [hunk 5]
```diff
@@ -144,6 +144,72 @@ When the vote passes on a release candidate, follow the steps below to complete
...
+
+1. Before running integration tests, you'll need the following information:
+    * Release version (example: `1.2.0`)
+    * Staging repo (example: `https://repository.apache.org/content/repositories/orgapachefluo-1023/`)
+        * Note: The staging repo changes with each release candidate.
```
Comment (ctubbsii) [inline pages/release-process.md]: Too wordy. Better to use a system property in the below profile, and have the user set it on the command line.
This can become:
```
1. Set the version number: $ export RCVERSION=1.2.0
2. Set the staging repository URL: $ export STAGINGREPO=https://...orgapachefluo-1023/
```

File: pages/release-process.md [hunk 6]
```diff
@@ -144,6 +144,72 @@ When the vote passes on a release candidate, follow the steps below to complete
...
+
+1. Add the following configuration to `~/.m2/settings.xml` to instruct Maven to build using the staging repo
+   for the release candidate. Also, make sure that the `fluoRC` profile is active while you are testing the release
+   but comment out `<activeProfiles>` section after you are done testing.
```
Comment (ctubbsii) [inline pages/release-process.md]: This is far too wordy and explanatory. Just put the command to create a temporary settings.xml file, which can be specified on the maven command line later:
```
3. Create a settings.xml file: $ cat <<EOF >/tmp/fluo-its-settings.xml
   ...
EOF
```

File: pages/release-process.md [hunk 7]
```diff
@@ -144,6 +144,72 @@ When the vote passes on a release candidate, follow the steps below to complete
...
+            <url>https://repository.apache.org/content/repositories/orgapachefluo-1023/</url>
+          </pluginRepository>
+        </pluginRepositories>
+      </profile>
+    </profiles>
+    <activeProfiles>
+      <activeProfile>fluoRC</activeProfile>
+    </activeProfiles>
+    ```
+1. Clone a project that uses Fluo like the [Fluo Recipes] project below:
```
Comment (ctubbsii) [inline pages/release-process.md]: No need for sentence. Just put the command.

File: pages/release-process.md [hunk 8]
```diff
@@ -144,6 +144,72 @@ When the vote passes on a release candidate, follow the steps below to complete
...
+    ```shell
+    $ git clone https://github.com/apache/fluo-recipes.git
+    ```
+    In addition to [Fluo Recipes], there are several projects containing example Fluo applications
+    that have integration tests:
+    * [Phrasecount] - `https://github.com/astralway/phrasecount.git`
+    * [Stresso] - `https://github.com/astralway/stresso.git`
+    * [Webindex] - `https://github.com/astralway/webindex.git`
+1. Run the integration tests using `mvn verify`.  Make sure to set the `fluo.version` to the release
```
Comment (ctubbsii) [inline pages/release-process.md]: Shorten to: `Build with: $ mvn -s /tmp/fluo-its-settings.xml clean verify -Dfluo.version=$RCVERSION`
Get rid of preceding paragraph which is redundant with the command.

File: pages/release-process.md [hunk 9]
```diff
@@ -144,6 +144,72 @@ When the vote passes on a release candidate, follow the steps below to complete
...
+   version.
+    ```shell
+    $ mvn clean verify -Dfluo.version=1.2.0
+    ```
+1. After you are done testing, remember to comment out `<activeProfiles>` in your `~/.m2/settings.xml`
```
Comment (ctubbsii) [inline pages/release-process.md]: No need for this if using a separate settings.xml file.

**Review by ctubbsii — COMMENTED** [review]
I like the idea, but I dislike how wordy it is. I think it should be far less descriptive, and much more "copy/paste". I've provided some suggested changes, but am flexible. I really do think it should be bare minimum, though.

**Comment by mikewalch** [comment]
@ctubbsii, I made a bunch of updates. Let me know what you think.

File: pages/release-process.md [hunk 10]
```diff
@@ -146,68 +146,108 @@ When the vote passes on a release candidate, follow the steps below to complete
...
+         </pluginRepositories>
+       </profile>
+     </profiles>
+     <activeProfiles>
+       <activeProfile>fluoRC</activeProfile>
+     </activeProfiles>
+   </settings>
+   EOF
+   ```
+1. If a new release candidate is made, update your staging repo.
```
Comment (ctubbsii) [inline pages/release-process.md]: Since these instructions show using `/tmp`, there's no reason to suspect that the previous file will still exist against the new RC. It'd probably be better to just have them start over. That could simplify some of this, because there'd be no reason to put the filename in a variable, or to use `\${env.RC_STAGING}` (could use `$RC_STAGING`) instead... or better yet... put most of the URL in the file, and just the number in the variable.

File: pages/release-process.md [hunk 11]
```diff
@@ -146,68 +146,108 @@ When the vote passes on a release candidate, follow the steps below to complete
...
+   ./bin/uno fetch fluo
+   ```
+1. Download release tarball and calculate hash
+   ```shell
+   wget -P downloads/ ${RC_STAGING}/org/apache/fluo/fluo/${RC_VERSION}/fluo-${RC_VERSION}-bin.tar.gz
+   shasum -a 256 downloads/fluo-${RC_VERSION}-bin.tar.gz
+   ```
+1. Set `FLUO_VERSION` and `FLUO_HASH` in `conf/uno.conf`.
+   ```shell
+   vim conf/uno.conf
```
Comment (ctubbsii) [inline pages/release-process.md]: A sed command would be better here than vim instructions.

File: pages/release-process.md [hunk 12]
```diff
@@ -146,68 +146,108 @@ When the vote passes on a release candidate, follow the steps below to complete
...
+   ```
+1. Set up Fluo and your shell
+   ```shell
+   ./bin/uno setup fluo
+   eval "$(./bin/uno env)"
+   ```
```
Comment (ctubbsii) [inline pages/release-process.md:227]: Is there more to test here, or just testing that uno can setup fluo?

File: pages/release-process.md [hunk 13]
```diff
@@ -146,68 +146,108 @@ When the vote passes on a release candidate, follow the steps below to complete
...
+
+### Run the integration tests of projects that use Fluo
```
Comment (ctubbsii) [inline pages/release-process.md]: I think this section should go after the phrasecount example with uno. It's really a separate test case, and it is breaking up the uno test case with phrasecount.

**Review by ctubbsii — APPROVED** [review]
I like this version better. I have additional ideas, but I think it might be easier if I submitted my own PR to make the changes in my head, rather than hold up your changes with my feedback.

File: pages/release-process.md (hunk 12 above)
Comment (mikewalch) [reply pages/release-process.md:227]: This is set up instructions for uno which is need to run example applications. I pushed a commit to make this more clear

File: pages/release-process.md (hunk 13 above)
Comment (mikewalch) [reply pages/release-process.md]: I reorganized it a little better.  I move this before the the uno / phrasecount examples.

File: pages/release-process.md [hunk 14]
```diff
@@ -144,6 +144,110 @@ When the vote passes on a release candidate, follow the steps below to complete
...
+   ./bin/uno fetch fluo
+   ```
+1. Download release tarball
+   ```shell
+   wget -P downloads/ ${RC_STAGING}/org/apache/fluo/fluo/${RC_VERSION}/fluo-${RC_VERSION}-bin.tar.gz
+   ```
+1. Set `FLUO_VERSION` and `FLUO_HASH` in `conf/uno.conf`.
+   ```shell
+   sed -i "s/export FLUO_VERSION=[^ ]*/export FLUO_VERSION=$RC_VERSION/" conf/uno.conf
+   sed -i "s/export FLUO_HASH=[^ ]*/export FLUO_HASH=$(shasum -a 256 downloads/fluo-${RC_VERSION}-bin.tar.gz | cut -d ' ' -f 1)/" conf/uno.conf
```
Comment (keith-turner) [inline pages/release-process.md:221]: this is nice

File: pages/release-process.md (hunk 10 above)
Comment (mikewalch) [reply pages/release-process.md]: I can do that `RC_STAGING` will just be the number (i.e `1023`)

Comment (mikewalch) [reply pages/release-process.md]: I like the idea of using the staging repo number. I pushed a commit that does this.

**Comment by mikewalch** [comment]
I pushed some more commits after this was approved.  Is it OK still I merge it?
